helm/vitess/templates/_vttablet.tpl

```diff
@@ -140,6 +142,87 @@ spec:
       shard: {{ $shardClean | quote }}
       type: {{ $tablet.type | quote }}
 
+{{ if eq $tablet.type "replica" }}
+---
+###################################
+# vttablet InitShardMaster Job
+###################################
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{ $shardName }}-init-shard-master
+spec:
+  backoffLimit: 0
+  template:
+    spec:
+      restartPolicy: OnFailure
+      containers:
+      - name: init-shard-master
+        image: "vitess/vtctlclient:{{$vitessTag}}"
+
+        command: ["bash"]
+        args:
+          - "-c"
+          - |
+            set -ex
+
+            SECONDS=0
+            TIMEOUT_SECONDS=600
+
+            # poll every 5 seconds to see if vtctld is ready
+            until vtctlclient -server vtctld.{{ $namespace }}:15999 ListAllTablets {{ $cellClean }} > /dev/null 2>&1; do 
+              if (( $SECONDS > $TIMEOUT_SECONDS )); then
+                echo "timed out waiting for vtctlclient to be ready"
+                exit 1
+              fi
+              sleep 5
+            done
+
+            until [ $TABLETS_READY ]; do
+              # get all the tablets in the current cell
+              cellTablets="$(vtctlclient -server vtctld.{{ $namespace }}:15999 ListAllTablets {{ $cellClean }})"
+              
+              # filter to only the tablets in our current shard
+              shardTablets=$( echo "$cellTablets" | awk 'substr( $5,1,{{ len $shardName }} ) == "{{ $shardName }}" {print $0}')
+
+              # check for a master tablet
+              masterTablet=$( echo "$shardTablets" | awk '$4 == "master" {print $1}')
+              if [ $masterTablet ]; then
+                  echo "'$masterTablet' is already the master tablet, exiting without running InitShardMaster"
+                  exit
+              fi
+
+              # count the number of newlines for the given shard to get the tablet count
+              tabletCount=$( echo "$shardTablets" | wc | awk '{print $1}')
+              
+              # check to see if the tablet count equals the expected tablet count
+              if [ $tabletCount == {{ $totalTabletCount }} ]; then
+                TABLETS_READY=true
+              else
+                # wait 5 seconds for vttablets to continue getting ready
+                sleep 5
+              fi
+
+              if (( $SECONDS > $TIMEOUT_SECONDS )); then
```

Comment:
Should this go inside the `else`, before the `sleep`? If the tablets are ready, we don't need to check for timeout.

Reply:
Done.